### PR TITLE
ara_frontend_nginx: add support for https/ssl

### PR DESCRIPTION
**Pull request by dmsimard**
This is rebased on top of https://github.com/ansible-community/ara-collection/pull/38 which isn't merged yet, will rebase on top of master when it is merged so all the commits belonging to it will go away.

### CODE REVIEW

**Comment by dmsimard** [comment]
It works: https://dashboard.zuul.ansible.com/t/ansible/build/c8e6119422ad4175a88aad1bef486218

Although the ssl certificate warning is quite noisy:
```
TASK [smoke-tests : include_tasks] *********************************************
task path: /home/zuul/.ara-tests/git/ara/tests/integration/roles/smoke-tests/tasks/main.yaml:22
included: /home/zuul/.ara-tests/git/ara/tests/integration/roles/smoke-tests/tasks/ara-ops.yaml for localhost

TASK [smoke-tests : Print normal data] *****************************************
task path: /home/zuul/.ara-tests/git/ara/tests/integration/roles/smoke-tests/tasks/ara-ops.yaml:20
/home/zuul/.ara-tests/virtualenv/lib64/python3.6/site-packages/urllib3/connectionpool.py:1020: InsecureRequestWarning: Unverified HTTPS request is being made to host 'ara.example.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
ok: [localhost] => {
    "msg": "normal data"
}

TASK [smoke-tests : Print confidential data] ***********************************
task path: /home/zuul/.ara-tests/git/ara/tests/integration/roles/smoke-tests/tasks/ara-ops.yaml:24
ok: [localhost] => {
    "censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"
}

TASK [smoke-tests : Record data with no type] **********************************
task path: /home/zuul/.ara-tests/git/ara/tests/integration/roles/smoke-tests/tasks/ara-ops.yaml:30
/home/zuul/.ara-tests/virtualenv/lib64/python3.6/site-packages/urllib3/connectionpool.py:1020: InsecureRequestWarning: Unverified HTTPS request is being made to host 'ara.example.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
changed: [localhost] => {
    "changed": true,
    "created": "2021-08-16T02:41:57.430370Z",
    "key": "notype",
    "msg": "Record created or updated in ARA",
    "playbook_id": 1,
    "type": "text",
    "updated": "2021-08-16T02:41:57.430427Z",
    "value": "text"
}
```

**Comment by Thulium-Drake** [comment]
Indeed it is noisy, but alas nothing we can really do about it (I think I'd rather have this noise in the logs then to try and work around it in the CI), this really should be solved by the end-user :)

File: roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    client_max_body_size {{ ara_nginx_client_max_body_size }};
+    access_log /var/log/nginx/{{ ara_api_fqdn }}_access.log;
+    error_log  /var/log/nginx/{{ ara_api_fqdn }}_error.log;
+
+    ssl on;
+    ssl_certificate {{ ara_api_frontend_ssl_cert }};
+    ssl_certificate_key {{ ara_api_frontend_ssl_key }};
+    ssl_session_cache   shared:SSL:10m;
+    ssl_session_timeout 10m;
+    ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;
```
Comment (Thulium-Drake) [inline roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2:33]: I'd remove TLS 1.0 and 1.1 from the list, they are severely outdated at this point

```
ssl_protocols TLSv1.2 TLSv1.3;
```

File: roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ssl_ciphers         HIGH:!aNULL:!MD5;
```
Comment (Thulium-Drake) [inline roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2:34]: The list below is very restrictive, but secure (for now at least ;-) )
```
ssl_ciphers EECDH+AESGCM:EDH+AESGCM;
```

**Review by Thulium-Drake — APPROVED** [review]
LGTM, but it seems this PR targets master, will that still work after merging #38 ?

**Comment by Thulium-Drake** [comment]
To allow Github to autoclose the issue

Fixes: #37

**Comment by dmsimard** [comment]
> LGTM, but it seems this PR targets master, will that still work after merging #38 ?

Yep, need to rebase against master and then push --force this branch like I just did.

File: roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2 (hunk 1 above)
Comment (dmsimard) [reply roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2]: Sure thing. Thanks !

File: roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2 (hunk 2 above)
Comment (dmsimard) [reply roles/ara_frontend_nginx/templates/ara-api-ssl.conf.j2]: Fixed, thanks !

**Comment by dmsimard** [comment]
Improved SSL ciphers like so: https://github.com/dmsimard/recordsansible.ara/commit/5ab92bf2dba4a93653b70f54c9bdfb41e00a8d13

Squashed into non-WIP commit ready to land.

**Comment by dmsimard** [comment]
recheck

**Pull request merged**